src/uct/tcp/tcp_ep.c

```diff
@@ -799,26 +827,32 @@ uct_tcp_ep_set_outstanding_zcopy(uct_tcp_iface_t *iface, uct_tcp_ep_t *ep,
     uct_tcp_ep_mod_events(ep, UCS_EVENT_SET_EVWRITE, 0);
 }
 
-static inline void uct_tcp_ep_am_send(uct_tcp_iface_t *iface, uct_tcp_ep_t *ep,
-                                      const uct_tcp_am_hdr_t *hdr)
+static inline ucs_status_t
+uct_tcp_ep_am_send(uct_tcp_iface_t *iface, uct_tcp_ep_t *ep,
+                   const uct_tcp_am_hdr_t *hdr)
 {
-    size_t sent_length;
+    ssize_t ret;
 
     ep->tx.length       = sizeof(*hdr) + hdr->length;
     iface->outstanding += ep->tx.length;
 
-    uct_tcp_ep_send(ep, &sent_length);
+    ret = uct_tcp_ep_send(ep);
+    if (ucs_unlikely(ret < 0)) {
+        return ret;
+    }
 
     uct_iface_trace_am(&iface->super, UCT_AM_TRACE_TYPE_SEND, hdr->am_id,
                        hdr + 1, hdr->length, "SEND: ep %p fd %d sent "
-                       "%zu/%zu bytes, moved to offest %zu",
-                       ep, ep->fd, ep->tx.offset, ep->tx.length, sent_length);
+                       "%zu/%zu bytes, moved to offest %zd",
+                       ep, ep->fd, ep->tx.offset, ep->tx.length, ret);
```

Comment:
looks like `ep->tx.offset` and `ret` parameters are mixed

Reply:
yep, they will print the same value, but it was done to align with other printouts to simplify `grep`ing
see, e.g. `uct_tcp_ep_progress_data_tx`

Reply:
let me clarify this. 
`ret` shows how many bytes were sent during out last send, while offset shows how many bytes of the message were sent in total so far.
Can't these 2 values be different at some point?

Reply:
> Can't these 2 values be different at some point?

yes, the can

Reply:
so need to fix it?

Reply:
`moved to offset %zd` -> `%zd bytes sent`?

Reply:
I'd change parameters, like
`"SEND: ep %p fd %d sent %zu/%zu bytes, moved to offest %zd", ep, ep->fd,  ret, ep->tx.length, ep->tx.offset);`
Also please fix `offest` to `offset`

Reply:
`%zu/%zu` but first `%zu` means total sent bytes

Reply:
ic now. Please just change   `moved to offest` -> `moved by offset` then

Reply:
sure

Reply:
`offest` - > `offset` ;)

Reply:
> `offest` - > `offset` ;)

fixed :)

Reply:
where?